### PR TITLE
feat: add `go-ipld-prime` instead of `go-ipld-cbor`

### PR DESCRIPTION
**Pull request by gsk967**
Closes: #40 

## Description

This pull request contains the changes related to adding `go-ipld-prime` instead of `go-ipld-cbor` 

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: utils/json.go [hunk 1]
```diff
@@ -50,3 +64,44 @@ func GetAttributeAsString(obj map[string]interface{}, attr string) (string, erro
...
+	// or if you need even more control, you could also write a function that conforms to the linking.BlockWriteOpener interface.
+	lsys.SetWriteStorage(&store)
+	// To create any links, first we need a LinkPrototype.
+	// This gathers together any parameters that might be needed when making a link.
+	// (For CIDs, the version, the codec, and the multihash type are all parameters we'll need.)
+	// Often, you can probably make this a constant for your whole application.
+	lp := cidlink.LinkPrototype{Prefix: cid.Prefix{
+		Version:  1,    // Usually '1'.
+		Codec:    0x71, // 0x71 means "dag-cbor" -- See the multicodecs table: https://github.com/multiformats/multicodec/
+		MhType:   0x13, // 0x20 means "sha2-512" -- See the multicodecs table: https://github.com/multiformats/multicodec/
```
[the remark this answers is not in the document]
Comment (gsk967) [reply utils/json.go:86]: I just take reference code from this link https://github.com/ipld/go-ipld-prime/blob/651b1cd5be900ae04b703bdff39b7727c26f36a6/linking/linkingExamples_test.go#L46 

But we can use any codec there

[the remark this answers is not in the document]
Comment (i-norden) [reply utils/json.go:86]: Let's use sha2-256

**Comment by gsk967** [comment]
@i-norden  can you review again , I address the pr comments

**Comment by ashwinphatak** [comment]
@gsk967 Please verify with system tests in the chiba clonk client.

**Comment by gsk967** [comment]
@ashwinphatak  I tested, it is working fine with chiba-clonk-client

**Pull request merged**